apis/flowcollector/v1beta2/flowcollector_types.go

```diff
@@ -373,6 +374,86 @@ type FlowCollectorIPFIXReceiver struct {
 	Transport string `json:"transport,omitempty"`
 }
 
+type FlowCollectorOpenTelemetryLogs struct {
+	// Set `enable` to `true` to send logs to Open Telemetry receiver.
+	//+kubebuilder:default:=true
+	Enable *bool `json:"enable,omitempty"`
+}
+
+type FlowCollectorOpenTelemetryMetrics struct {
+	// Set `enable` to `true` to send metrics to Open Telemetry receiver.
+	//+kubebuilder:default:=true
+	Enable *bool `json:"enable,omitempty"`
+
+	// Prefix added to each metric name
+	// +optional
+	Prefix string `json:"prefix,omitempty"`
+
+	// How often should metrics be sent to collector
+	// +kubebuilder:default:="20s"
+	PushTimeInterval *metav1.Duration `json:"pushTimeInterval,omitempty"`
+
+	// Time duration of no-flow to wait before deleting data item
+	// +kubebuilder:default:="2m"
+	ExpiryTime *metav1.Duration `json:"expiryTime,omitempty"`
+}
+
+type FlowCollectorOpenTelemetryTraces struct {
+	// Set `enable` to `true` to send traces to Open Telemetry receiver.
+	//+kubebuilder:default:=true
+	Enable *bool `json:"enable,omitempty"`
+
+	// Separate span for each prefix listed
+	//+optional
+	SpanSplitter []string `json:"spanSplitter,omitempty"`
+}
+
+// Add more exporter types below
+type ConnectionType string
+
+const (
+	httpConnection ConnectionType = "http"
+	grpcConnection ConnectionType = "grpc"
+)
+
+type FlowCollectorOpenTelemetry struct {
+	// Address of the Open Telemetry receiver
+	// +kubebuilder:default:=""
+	TargetHost string `json:"targetHost"`
+
+	// Port for the Open Telemetry receiver
+	TargetPort int `json:"targetPort"`
+
+	// Type of Open Telemetry connection. The available options are `http` and `grpc`.
+	// +unionDiscriminator
+	// +kubebuilder:validation:Enum:="http";"grpc"
+	// +kubebuilder:default:="grpc"
+	Type ConnectionType `json:"type"`
+
+	// Headers to add to messages (optional)
+	// +optional
+	Headers map[string]string `json:"headers,omitempty"`
+
+	// TLS client configuration.
+	// +optional
+	TLS ClientTLS `json:"tls"`
+
+	// Rules to transform flow logs to Opentelemetry-conformant format.
+	Rules []api.GenericTransformRule `json:"rules,omitempty"`
```

Comment:
Should we expose rules or hardcode them ?

See https://github.com/netobserv/flowlogs-pipeline/pull/531 for FLP implementation